### PR TITLE
Maneuver4 2

### PR DESCRIPTION
**Pull request by kdemski**
still needs comments and update from Development,

### CODE REVIEW

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java [hunk 1]
```diff
@@ -120,6 +120,28 @@ public void onEvent(@Nullable TimelineElement element, TimelineEvent event, DJIE
     }
 
     /**
+     * loads the manuever to the drone
+     * @param data takes in mFlightPlanData
+     */
+    public void loadManeuver(FlightPlanData data){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: the base load timeline mission function should work for maneuver as well, just pass in an array of one

Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: incase in the future we want multi mission maneuvers with different mission types

File: app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java [hunk 2]
```diff
@@ -294,4 +293,98 @@ private void vecToMwp(double lon, double lat, float alt, MissionVector vec){
...
+        MissionVector vec = new MissionVector(vecNum, vecTyp, distance, vertDist, direction, heading, radius);
+        return vec;
+    }
+
+    /**
+     * loads the maneuver into mFlightPlanData, uses a case structure to analyze a string maneuverName, and pick out the correct stringArray from resources R
+     * @param maneuverName
+     * @return loaded true if a maneuver has been found and loaded
+     */
+    public boolean loadManeuver(String maneuverName){
```
Comment (owen927) [inline app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java]: serves better as a class on its own, with supporting functions to return selected maneuvers, see my inspection data for non hardcoded implementations that allows expansion and changes to maneuvers in the future without affecting the code

File: app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java [hunk 3]
```diff
@@ -8,8 +8,8 @@
 import android.database.Cursor;
```
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java]: add all maneuvers to a variable in the loader constructor, and then never access the resources

File: app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java [hunk 4]
```diff
@@ -840,17 +850,37 @@ private void startStopMission(){
...
+        MissionLoader.maneuverRunning = true;
+        hideSuspendButton(); //user shouldn't be able to use suspend button during a maneuver, (also it causes a crash)
+        removeManeuverGridView();
+        startButtonOn();
+    }
 
+    /**
+     * startMission checks if flight data is loaded in inspection session, then loads and start the selected mission.
+     */
     private void startMission(){
```
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: put a start maneuver check at the top of the startMission and try to integrate startManeuver into StartMission and delete the startManeuverCode

File: app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java [hunk 5]
```diff
@@ -1174,6 +1228,92 @@ public void onClick(DialogInterface dialog, int id) {
...
+                    }
+                }else{
+                    ToastUtils.setResultToToast("don't have a string for that pic bro");
+                }
+            }
+        });
+    }
+
+    /**
+     * toggles the maneuver grid view to either appear or gone
```
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: if not connected to drone, then user shouldn't be able to start a maneuver,. i.e. setEnabled False setEnabled()

Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: make new class for the maneuvers in memory

Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/controller/MainActivity.java]: done

File: app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java (hunk 1 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionManager/TimeLineMissionControl.java]: updated

File: app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java (hunk 2 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java]: Updated

File: app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java (hunk 3 above)
Comment (kdemski) [inline app/src/main/java/com/aeriosense/projecttalon/missionLoader/MissionLoader.java]: updated

**Pull request closed without merging**